src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -62,7 +62,8 @@ static grpc_security_status ssl_create_security_connector(
     }
   }
   status = grpc_ssl_channel_security_connector_create(
-      exec_ctx, call_creds, &c->config, target, overridden_target_name, sc);
+      exec_ctx, creds, call_creds, &c->config, target, overridden_target_name,
```

Comment:
Can you please do the same for the fake connector? I'm wondering whether we shouldn't add a NULL-check in the comparison function as a way to force this field to be set on all security connector implementations... I'm worried that we're going to add a new type of connector, forget this check, and have a security hole... For httpcli_connector, we could set a static one. What do you think?

Reply:
Instead of trying to generalize this further, I've gone the opposite way: I've moved the credential references into the SSL security connectors, since they're the only ones that actually need it.  This is now purely an implementation detail of the individual security connector implementation: those that need the refs can keep them, and those that don't will not have to worry about them.

Reply:
That's not actually true. We have some internal ones that need exactly the same checks: the "fake" and "httpcli" ones are the outliers here. Logically the credentials encapsulate user configuration and this config has to be the same for 2 security connectors to be equal.

Reply:
Right, I understand that there will be other implementations that need to do this.  What I was saying is that we should make that the responsibility of the implementations that need this to do this check, rather than assuming that they will all need to do so.  In other words, our internal implementations can do the same thing that the SSL security connectors are now doing in this PR.

If you feel strongly, I can go back the other way.  I guess it just feels a little ugly to have to create a fake credentials object for the cases where no credentials are needed.

Reply:
As per our conversation, changed to enforce this in the base class.  PTAL.